packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -20,6 +21,175 @@ import 'scroll_position.dart';
+  }
+}
+
+/// A delegate that supplies infinite children for [ListWheelScrollView] by
+/// looping an explicit list.
+///
+/// [ListWheelScrollView] lazily construct its children to avoid creating more
+/// children than are visible through the [Viewport]. This delegate provides
+/// children using an explicit list, which is convenient but reduces the benefit
+/// of building children lazily.
+///
+/// In general building all the widgets in advance is not efficient. It is
+/// better to create a delegate that builds them on demand using
+/// [ListWheelChildBuilderDelegate] or by subclassing [ListWheelChildDelegate]
+/// directly.
+///
+/// This class is provided for the cases where either the list of children is
+/// known well in advance (ideally the children are themselves compile-time
+/// constants, for example), and therefore will not be built each time the
+/// delegate itself is created, or the list is small, such that it's likely
+/// always visible (and thus there is nothing to be gained by building it on
+/// demand). For example, the body of a dialog box might fit both of these
+/// conditions.
+class ListWheelChildLoopingListDelegate extends ListWheelChildDelegate {
+  /// Constructs the delegate from a concrete list of children.
+  ListWheelChildLoopingListDelegate({@required this.children}) : assert(children != null);
+
+  /// The list containing all children that can be supplied.
+  final List<Widget> children;
+
+  @override
+  int get estimatedChildCount => null;
+
+  @override
+  int trueIndexOf(int index) => index % children.length;
+
+  @override
+  Widget build(BuildContext context, int index) {
+    if (children.isEmpty)
+      return null;
+    return children[index % children.length];
+  }
+
+  @override
+  bool shouldRebuild(covariant ListWheelChildLoopingListDelegate oldDelegate) {
+    return children != oldDelegate.children;
+  }
+}
+
+/// A delegate that supplies children for [ListWheelScrollView] using a builder
+/// callback.
+///
+/// [ListWheelScrollView] lazily construct its children to avoid creating more
+/// children than are visible through the [Viewport]. This delegate provides
+/// children using an [IndexedWidgetBuilder] callback, so that the children do
+/// not even have to be built until they are displayed.
+class ListWheelChildBuilderDelegate extends ListWheelChildDelegate {
+  /// Constructs the delegate from a builder callback.
+  ListWheelChildBuilderDelegate({
+    @required this.builder,
+    this.childCount,
+  }) : assert(builder != null);
+
+  /// Called lazily to build children.
+  final IndexedWidgetBuilder builder;
+
+  /// {@template flutter.widgets.wheelList.childCount}
+  /// If non-null, [childCount] is the maximum number of children that can be
+  /// provided, and children are available from 0 to [childCount] - 1.
+  ///
+  /// If null, then the lower and upper limit are not known. However the [builder]
+  /// must provide children for a contiguous segment and returning null at
```

Comment:
Suggestion: split this into two sentences, replacing 'and' with a period.

Then, clarify the beginning of the next sentence:
'If the [builder] returns null at some index, the segment terminates there'.